.github/workflows/ci.yml

```diff
@@ -29,34 +29,53 @@ jobs:
   # build binaries. use release for staging
   # this requires check and test
   build_binaries:
-    name: Build binaries for Target (${{ matrix.rust_target }}) on ${{ matrix.os }}
+    name: Build ${{ matrix.binary }} for ${{ matrix.target }} (${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        rust_target:
+        target:
 #          - aarch64-unknown-linux-musl
 #          - armv7-unknown-linux-gnueabihf
           - x86_64-apple-darwin
           - x86_64-unknown-linux-musl
         rust: [stable]
+        binary: [fluvio]
         include:
-#          - rust_target: aarch64-unknown-linux-musl
-#            os: ubuntu-latest
-#            binaries: [fluvio]
-#          - rust_target: armv7-unknown-linux-gnueabihf
-#            os: ubuntu-latest
-#            binaries: [fluvio]
-          - rust_target: x86_64-apple-darwin
-            os: macos-latest
-            binaries: [fluvio]
-          - rust_target: x86_64-unknown-linux-musl
-            os: ubuntu-latest
-            binaries: [fluvio, fluvio-run, flv-test]
+#          - os: ubuntu-latest
+#            rust: stable
+#            target: aarch64-unknown-linux-musl
+#            binary: fluvio
+#            run: make build-cli
+#          - os: ubuntu-latest
+#            rust: stable
+#            target: armv7-unknown-linux-gnueabihf
+#            binary: fluvio
+#            run: make build-cli
+          - os: macos-latest
+            rust: stable
+            target: x86_64-apple-darwin
+            binary: fluvio
+            run: make build-cli
```

Comment:
this should only contain matrix configuration not how to perform steps

Reply:
The problem is that the old way is really ugly and unweildy, with several disadvantages:

- Each instance of the build job has X steps, one for each binary that might be built, but all except 1 of those are disabled, making the job screen really cluttered and ugly, e.g. here on a target where we only want `fluvio`:

![image](https://user-images.githubusercontent.com/4210949/125856185-508ada44-cc9d-4e6b-9977-3714eddcdee5.png)

- This problem gets worse for every new binary we might want to add to the workflow

When we set up the job this way, it takes much less code and it also displays in the nicest way in the actions UI. I really don't see any disadvantages to doing things this way, it makes it much easier.

Reply:
This is opposed to the newer, flatter layout where each job is only instantiated with steps that it actually uses:

![image](https://user-images.githubusercontent.com/4210949/125856488-c7827869-01de-4352-bfc1-5c32a6803415.png)

Reply:
that's implementation details.  important point is that `matrix` is configuration.  It should not contain how to perform configuration.